server/api.ts

```diff
@@ -30,3 +44,13 @@ export interface SignInRequest {
   password: string;
 }
 export type SignInResponse = Pick<User, 'email' | 'firstName' | 'lastName' | 'username' | 'id'>;
+export type GetUserByEmailRequest = { emailId: string };
+export interface GetUserByEmailResponse {
+  user: User;
+}
+export type GetUserByUserNameRequest = {
+  username: string;
+};
+export interface GetUserByUserNameResponse {
+  user: User;
+}
```

Comment:
I don't think we want these actually, the sign in/up are enough, unless you have a specific use case in mind?

Reply:
you are right, I removed them.